app/services/katello/registration_manager.rb

```diff
@@ -306,6 +306,7 @@ def remove_host_artifacts(host, clear_content_facet: true)
           host.content_facet.applicable_errata = []
           host.content_facet.uuid = nil
           host.content_facet.content_view_environments = []
+          host.content_facet.content_source = ::SmartProxy.pulp_primary
```

Comment:
Think this is worth a test? Perhaps in `test/services/katello/registration_manager_test.rb`